stablehlo/dialect/TypeInference.h

```diff
@@ -219,10 +262,63 @@ LogicalResult inferTriangularSolveOp(
 LogicalResult inferWhileOp(Optional<Location> location, ValueRange operand,
                            SmallVectorImpl<Type>& inferredReturnTypes);
 
+//===----------------------------------------------------------------------===//
+// Verifiers for ops.
+//===----------------------------------------------------------------------===//
+
+LogicalResult verifyAllReduceOp(Optional<Location> location, Value operand,
+                                DenseIntElementsAttr replicaGroups,
+                                Region& computation);
+
+LogicalResult verifyBitcastConvertOp(Optional<Location> location, Value operand,
+                                     Value result);
+
+LogicalResult verifyBroadcastOp(Optional<Location> location,
+                                DenseIntElementsAttr sizes);
+
+LogicalResult verifyBroadcastInDimOp(Optional<Location> location, Value operand,
+                                     DenseIntElementsAttr broadcastDimensions,
+                                     Value result);
+
+LogicalResult verifyClampOp(Optional<Location> location, Value min,
+                            Value operand, Value max);
+
+LogicalResult verifyCollectivePermuteOp(Optional<Location> location,
+                                        DenseIntElementsAttr sourceTargetPairs);
+
+LogicalResult verifyDynamicBroadcastInDimOp(
+    Optional<Location> location, Value operand, Value outputDimensions,
+    DenseIntElementsAttr broadcastDimensions,
+    Optional<DenseIntElementsAttr> knownExpandingDimensions,
+    Optional<DenseIntElementsAttr> knownNonexpandingDimensions, Value result);
+
+LogicalResult verifyDynamicReshapeOp(Optional<Location> location,
+                                     Value outputShape, Value result);
+
+LogicalResult verifyDynamicSliceOp(Optional<Location> location, Value operand,
+                                   ValueRange startIndices,
+                                   DenseIntElementsAttr sliceSizes);
+
+LogicalResult verifyGetTupleElementOp(Optional<Location> location,
+                                      Value operand, uint32_t indexVal,
```

Comment:
Should this be `int32_t`, similarly to `inferAllToAllOp`? Same feedback for `verifyIotaOp`, `verifyReduceScatterOp`, as well as existing `inferBatchNormFooOp` and `verifySortOp`.